torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,479 @@
+            source = os.path.join(source_folder, file_name)
+            destination = os.path.join(destination_folder, file_name)
+            if os.path.isfile(source):
+                shutil.move(source, destination)
+
+        shutil.rmtree(source_folder)
+        dest_split = os.path.split(destination_folder)
+        if len(modality.split("_")) == 3:
+            rename_dest = os.path.join(dest_split[0], "s2_temporal_subset")
+        if len(modality.split("_")) == 2:
+            rename_dest = os.path.join(
+                dest_split[0],
+                dest_split[1].split("_")[0] + "_" + dest_split[1].split("_")[1],
+            )
+        if "_" not in modality:
+            rename_dest = os.path.join(dest_split[0], dest_split[1].split("_")[0])
+        os.rename(destination_folder, rename_dest)
+
+    def _convert_to_color(
+        self, arr_2d: Tensor, cmap: dict[int, tuple[int, int, int]]
+    ) -> "np.typing.NDArray[np.uint8]":
+        """Numeric labels to RGB-color encoding.
+
+        Args:
+            arr_2d: 2D array to be colorized
+            cmap: colormap to use when mapping the labels
+
+        Returns:
+            3D colored image
+        """
+        arr_3d = np.zeros((arr_2d.shape[0], arr_2d.shape[1], 3), dtype=np.uint8)
+
+        for c, i in cmap.items():
+            m = arr_2d == c
+            arr_3d[m] = i
+        return arr_3d
+
+    def _get_bands(
+        self,
+        image: "np.typing.NDArray[np.float32]",
+        all_bands: Sequence[str],
+        select_bands: Sequence[str],
+    ) -> "np.typing.NDArray[np.float32]":
+        """Filters the bands for a given set of bands.
+
+        Args:
+            image: the image whose bands to be filtered
+            all_bands: all Sentinel-2 bands
+            select_bands: bands to filter
+
+        Returns:
+            the raster image with filtered bands
+        """
+        bands = [
+            all_bands.index(select_bands)
+            if isinstance(select_bands, str)
+            else select_bands
+            for select_bands in select_bands
+        ]
+        return image[bands]
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample image-mask pair returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        image = rearrange(sample["image"], "h w c -> w c h")
+        image = image.numpy()
+        mask = sample["mask"].squeeze()
+        color_mask = self._convert_to_color(mask, cmap=self.mask_cmap)
+
+        # Land cover
+        if np.all(np.isin(image, np.arange(0, 110, 10))) and image.shape[-1] == 1:
```

Comment:
Feels like there should be a better way to check whether or not the image is land cover. What if the image is really dark (all low numbers)? Could be a false positive

Reply:
Refactored it a bit but your scenario still holds. Could reach the metadata and statistics to add more checks on resolution and mean/std values but it could introduce some overhead. Do you think this is an edge scenario or just needs to be covered?

Reply:
Does `self.modality` tell you what you need to know? I think that's a lot safer than relying on image values/shapes.

Reply:
Not explicitly. It is a list of modalities set by the user. I can expand on the check with `and esa_wc in self.modality` but it will not clear out the inter-confusion.

Reply:
Hmm, correct me if I'm wrong but the current plotting method doesn't really work if multiple modalities are selected, correct? Maybe it would make more sense to have a for loop over `self.modalities` and use that to index which channels of the image are which modality. Then you can add a plot for each.